web/src/message_view_header.js

```diff
@@ -191,6 +195,9 @@ export function open_search_bar_and_close_narrow_description() {
     // otherwise fill the input field with the text operators for
     // the current narrow.
     if ($("#search_query").text() === "") {
+        // TODO: this is happening too late, or at least the typeahead
+        // `source` happens before this, so it doesn't exclude the
+        // right things from the dropdown options.
```

Comment:
I think this might be a consequence of an existing bug in the redesign actually https://github.com/zulip/zulip/pull/24345#issuecomment-1729306042